0.1.2 warehousing using dbt, duckdb and iceberg

redpheonixx opened the pull request:
---
Title: Boiler Plate Code for DBT Transformations

Assignees: Amit Singh Labels: Transformation layer

Description:

This pull request introduces a boilerplate code for DBT transformations using the medallion architecture. The transformations are structured across three layers:

Bronze Layer: Initial raw data ingestion and storage.

Silver Layer: Intermediate transformations to refine and standardize data.

Gold Layer: Final transformations for analytical purposes, providing clean and consumable data.

The implementation utilizes the Iceberg REST catalog to manage the datasets and transformations efficiently. This setup aims to streamline the development and maintenance of DBT models, ensuring a robust and scalable data pipelin

---

#45 

wrote dbt transformation layer as per medallion architecture 
 
dbt's low learning curve and SQL-based workflow empower data teams without advanced programming knowledge to build and maintain transformations. This makes it a better fit for companies looking for agility and simplicity in data pipelines.

redpheonixx committed:
folder structure for dbt transformation

redpheonixx committed:
adding python scripts for execution and orchestrator.py to trigger transformation

redpheonixx committed:
adding makefile for 0.1.2

tusharchou commented on local-data-platform/local_data_platform/makefile:
@redpheonixx what issue is this associated with?

redpheonixx replied:
#57 
following below blog for makefile 
https://motherduck.com/blog/duckdb-python-e2e-data-engineering-project-part-1/

tusharchou replied:
@redpheonixx what happened to this?

redpheonixx committed:
Merge branch 'tusharchou:main' into main

redpheonixx committed:
adding sqlachemy package

tusharchou committed:
V0.1.1 (#58)

tusharchou committed:
0.1.1 BigQuery Release  (#46)

* Create publish.yml

* Update publish.yml

tusharchou committed:
pypi release through github actions (#60)

* Create publish.yml

* Update publish.yml

* Update publish.yml

tusharchou committed:
Create manual.yml

tusharchou committed:
Update publish.yml (#62)

tusharchou committed:
Update publish.yml (#63)

tusharchou committed:
Update pyproject.toml (#64)

tusharchou committed:
Update publish.yml (#65)

* Update publish.yml

* Update publish.yml

* Update publish.yml

brmhastra committed:
Brmhastra patch 1 (#67)

* Update publish.yml

* Update publish.yml

* Update publish.yml

* Update publish.yml

updated yaml file to copy distribution o/p from build to root directory

* Update publish.yml

added detailed copy from /ldf/dist to gihub/wo../dist

---------

Co-authored-by: Tushar Choudhary <151359025+tusharchou@users.noreply.github.com>

brmhastra committed:
Brmhastra patch 1 (#68)

* Update publish.yml

* Update publish.yml

* Update publish.yml

* Update publish.yml

updated yaml file to copy distribution o/p from build to root directory

* Update publish.yml

added detailed copy from /ldf/dist to gihub/wo../dist

* Update publish.yml

added a code for creating dist directory

---------

Co-authored-by: Tushar Choudhary <151359025+tusharchou@users.noreply.github.com>

brmhastra committed:
Brmhastra patch 1 (#69)

* Update publish.yml

* Update publish.yml

* Update publish.yml

* Update publish.yml

updated yaml file to copy distribution o/p from build to root directory

* Update publish.yml

added detailed copy from /ldf/dist to gihub/wo../dist

* Update publish.yml

added a code for creating dist directory

* Update publish.yml

relocating dots from ./github to /.github

---------

Co-authored-by: Tushar Choudhary <151359025+tusharchou@users.noreply.github.com>

mrutunjay-kinagi committed:
GitHub workflow for pypi release (#70)

* Update publish.yml

* Update pyproject.toml

mrutunjay-kinagi committed:
GitHub workflow for pypi release (#71)

* Update publish.yml

* Update pyproject.toml

* Update publish.yml

mrutunjay-kinagi committed:
Update publish.yml (#72)

mrutunjay-kinagi committed:
Update publish.yml (#73)

mrutunjay-kinagi committed:
Release v1.1 changes (#74)

mrutunjay-kinagi committed:
Github action setup for release v1.1 (#76)

* Release v1.1 dist changes

* Release v1.1 publish.yml changes

* Release v1.1 publish.yml changes

redpheonixx committed:
Update publish.yml

redpheonixx committed:
Merge pull request #77 from tusharchou/redpheonixx-patch-1

Update publish.yml

redpheonixx committed:
Merge pull request #61 from tusharchou/tusharchou-patch-4

Create manual.yml

redpheonixx committed:
Update publish.yml (#78)

redpheonixx committed:
Update publish.yml

redpheonixx committed:
Merge pull request #79 from tusharchou/redpheonixx-patch-3

Update publish.yml

redpheonixx committed:
Update publish.yml (#80)

redpheonixx committed:
Update pyproject.toml (#81)

mrutunjay-kinagi committed:
Release v1.1 fix (#84)

* Release v1.1 dist changes

* Release v1.1 publish.yml changes

* Release v1.1 publish.yml changes

* Release v1.1

tusharchou commented:
Raising a pull request (PR) from a fork's main branch to a repository's main branch is generally discouraged for several reasons:

1. **Complexity and Confusion**: Using the main branch of your fork for PRs can create confusion if you make other changes or updates to your fork's main branch. It becomes harder to track what specific changes the PR is addressing.

2. **Continuous Integration Issues**: Many projects have automated checks and continuous integration (CI) workflows that run on branches. If you use the main branch for PRs, these workflows might get triggered unnecessarily, potentially causing conflicts or errors.

3. **Maintenance Difficulty**: It's easier to maintain a clean and organized history if you create a separate feature branch for each set of changes. This approach avoids cluttering the main branch and makes it simpler to manage and merge changes.

4. **Conflict Management**: Keeping changes in isolated branches helps in managing conflicts better. If your main branch is used for a PR and it diverges significantly from the upstream main branch, resolving conflicts can become more complicated.

5. **Best Practices**: Following best practices, like creating feature branches for specific changes or issues, improves collaboration and code quality. It also aligns with how most open-source projects and development teams operate.

### Best Practice Workflow:
1. **Create a Feature Branch**:
   ```sh
   git checkout -b feature/your-feature-name
   ```

2. **Make Your Changes and Commit**:
   ```sh
   git commit -m "Describe your changes"
   ```

3. **Push Your Feature Branch to Your Fork**:
   ```sh
   git push origin feature/your-feature-name
   ```

4. **Open a PR from Your Feature Branch to the Upstream Main Branch**:
   - Go to the repository on GitHub.
   - Open a PR from `your-fork:feature/your-feature-name` to `upstream:main`.

This approach keeps your development process clean and organized. It ensures that the main branches remain stable and only contain code that is ready for production.

Feel free to ask if you need more details or help with Git workflows! 🚀✨

redpheonixx committed:
dbt transformations commit in new branch

redpheonixx committed:
Merge branch 'main' of https://github.com/redpheonixx/local-data-platform

redpheonixx committed:
Merge branch 'main' into 45-014-dbt-transformation-layer

redpheonixx committed:
Merge pull request #1 from redpheonixx/45-014-dbt-transformation-layer

dbt transformations commit in new branch

mrutunjay-kinagi committed:
0.1.1 Pytest Added for BigQuery Source  (#86)

* fixed bug warehouse uri

Thu Oct 24 9:25 PM IST

* added logging to CSV.get()

* supported big query ts format

* refactored parameter to config from catalog to reduce confusion

* Fixed bug of logger in GCP

* replaced local path with dynamic path

* replaced local path with dynamic path

* replaced local path with dynamic path

* demo

* demo

* Release v1.1 dist changes

* Release v1.1 publish.yml changes

* Release v1.1 publish.yml changes

* added a class implementation for github issue hoping it is useful for guiding users to resolution ETAs eventually

* added a exception for PlanNotFound to ask users to raise issues on the repository for resolution

* Updated overview and milestones. Added directory structure under technical specifications.

* Updated components in technical speciifcations

* added testing for BigQueryToCSV.extract()

* added testing for BigQueryToCSV.extract()

* added testing for Iceberg.get()

* Release v1.1

* Release v1.1 bug fix

* Release v1.1 bug fix

* Release v1.1 bug fix

* Pytest Added for BigQuery Source

---------

Co-authored-by: Tushar Choudhary <151359025+tusharchou@users.noreply.github.com>

redpheonixx committed:
Merge branch 'tusharchou:main' into main

rakhioza07 commented on local-data-platform/dist/local_data_platform-0.1.0.tar.gz:
Please don't add tar.gz, whl or heavy source files as a part of code commits

Only consider code, sample data and config file, scripts for PRs

tusharchou replied:
@redpheonixx please respond , thank you @rakhioza07

rakhioza07 commented on local-data-platform/local_data_platform/transformations/dbt_models/bronze/bronze.sql:
Consider moving all SQL files to a central place

Consider keeping code, data, queries, configs in seperate directory structures

tusharchou replied:
great idea @rakhioza07 , @redpheonixx look up dbt package

rakhioza07 commented on local-data-platform/local_data_platform/transformations/silver_layer.py:
This script needs to be OOP

tusharchou replied:
great idea @rakhioza07 , i will pick it up in next PR

rakhioza07 commented on local-data-platform/poetry.lock:
Does this file need to be a part of the commit? If no, please mention in gitignore

tusharchou replied:
I think it does @redpheonixx please confirm

rakhioza07 commented on local-data-platform/real_world_use_cases/near_data_lake/tmp.py:
```diff
@@ -0,0 +1,34 @@
+
+p=Path('C:/Users//singsina//Desktop//local-data-platform//local-data-platform//tmp//warehouse//')
+print(p)
+p_path="file:///"+str(p)
```
Use f-strings for string manipulation

rakhioza07 reviewed (changes requested):
Please fix review comments and move into draft PR until changes are made.

Great work on exception handling and clean coding efforts!

tusharchou commented on .github/workflows/manual.yml:
```diff
@@ -0,0 +1,32 @@
+# This is a basic workflow that is manually triggered
```
please take a pull from main to sync with the latest changes @redpheonixx sorry for the late review

tusharchou commented on .gitignore:
pull changes please

tusharchou commented on local-data-platform/README.md:
<img width="681" height="651" alt="Screenshot 2025-08-02 at 11 58 17 PM" src="https://github.com/user-attachments/assets/174f783e-9050-4ff4-a540-6acb17f0221e" />
Please update proper documentation as discussed on call

tusharchou commented on local-data-platform/pyproject.toml:
seems wrong we did it

tusharchou reviewed (approved):
Thank you. for everything @redpheonixx You have been a core contributor and I wish you all the best in your career! 🚀

rakhioza07 reviewed (approved):
Unseen/unreviewed changes, approving as directed by @tusharchou . Please rollback in case of failed deployment.

mrutunjay-kinagi commented:
@redpheonixx, Please resolve conflicts and update the PR.

cc @tusharchou
